### PR TITLE
Fixes and Improves iptables cleanup for pure iptables based proxier.

### PR DESCRIPTION
**Pull request by aveshagarwal**
Fixes and Improves iptables cleanup for pure iptables based proxier. Otherwise, pure iptables based proxier leaves rules that can only be cleaned up manually.

### CODE REVIEW

**Comment by ncdc** [comment]
@kubernetes/rh-networking

**Comment by ArtfulCoder** [comment]
I took a look.
assigning to @thockin since he was deeply involved with the review of the code.

**Comment by aveshagarwal** [comment]
Would appreciate any feedback.

**Comment by thockin** [comment]
Sorry for teh delay - we're a little buried in 1.1 prep right now - I have
not lost this, and I will get to it.

File: pkg/proxy/iptables/proxier.go [hunk 1]
```diff
@@ -206,15 +206,36 @@ func NewProxier(ipt utiliptables.Interface, exec utilexec.Interface, syncPeriod
 // It returns true if an error was encountered. Errors are logged.
 func CleanupLeftovers(ipt utiliptables.Interface) (encounteredError bool) {
 	//TODO: actually tear down all rules and chains.
-	args := []string{"-j", "KUBE-SERVICES"}
-	if err := ipt.DeleteRule(utiliptables.TableNAT, utiliptables.ChainOutput, args...); err != nil {
+	args := []string{"-m", "comment", "--comment", "kubernetes service portals"}
+	if err := ipt.DeleteRule(utiliptables.TableNAT, utiliptables.ChainOutput, append(args, "-j", string(iptablesServicesChain))...); err != nil {
```
Comment (thockin) [inline pkg/proxy/iptables/proxier.go]: why append() here rather than just put the -j string(...) in the `args :=` statement?

**Comment by thockin** [comment]
thanks!

**Comment by aveshagarwal** [comment]
@thockin Fixed and updated PR.

File: pkg/proxy/iptables/proxier.go (hunk 1 above)
Comment (aveshagarwal) [inline pkg/proxy/iptables/proxier.go]: Fixed it.

**Pull request merged**